frontend/app/components/session-timeout.tsx

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { type IIdleTimerProps, useIdleTimer } from 'react-idle-timer';
+
+import { Button } from './buttons';
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from '~/components/dialog';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('gcweb');
+
+export interface SessionTimeoutProps extends Pick<IIdleTimerProps, 'promptBeforeIdle'>, Pick<IIdleTimerProps, 'timeout'> {}
+
+const SessionTimeout = ({ promptBeforeIdle, timeout }: SessionTimeoutProps) => {
+  const { t } = useTranslation(i18nNamespaces);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState('');
+
+  const handleOnIdle = () => {
+    window.location.href = '/auth/logout';
+  };
+
+  const handleOnIdleContinueSession = () => {
+    setModalOpen(false);
+    reset();
+  };
+
+  const { reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    onPrompt: () => setModalOpen(true),
+    promptBeforeIdle: promptBeforeIdle ?? 5 * 60 * 1000, //5 minutes
+    timeout: timeout ?? 20 * 60 * 1000, //20 minutes
+  });
+
+  const tick = useCallback(() => {
+    const minutes = Math.floor(getRemainingTime() / 60000);
+    const seconds = Math.floor((getRemainingTime() / 1000) % 60).toFixed(0);
+    setTimeRemaining(minutes + ':' + (parseInt(seconds) < 10 ? '0' : '') + seconds);
+  }, [getRemainingTime]);
+
+  useEffect(() => {
+    setInterval(tick, 1000);
+  }, [tick]);
+
+  return (
+    <div>
+      <Dialog open={modalOpen && timeRemaining.length > 0}>
+        <DialogContent className="sm:max-w-md">
+          <DialogHeader>
+            <DialogTitle>{t('session-timeout.header')}</DialogTitle>
+          </DialogHeader>
+          {t('session-timeout.description', { timeRemaining })}
+          <DialogFooter>
+            <Button id="end-session-button" variant="primary" onClick={handleOnIdle}>
+              {t('session-timeout.end-session')}
+            </Button>
+            <Button id="continue-session-button" variant="default" onClick={handleOnIdleContinueSession}>
```

Comment:
variant should be primary and use a smaller size

Reply:
updated